ivy/functional/backends/jax/elementwise.py

```diff
@@ -304,6 +304,10 @@ def log10(x: JaxArray, /, *, out: Optional[JaxArray] = None) -> JaxArray:
 
 
 def log1p(x: JaxArray, /, *, out: Optional[JaxArray] = None) -> JaxArray:
+    if x.dtype in ["complex64", "complex128"]:
+        # For certain complex numbers, jnp.log1p(z) != jnp.log(1+z)
```

Comment:
That's a bit strange, were there any particular category of examples which were leading to these issues (i.e. very small values, very large values, with a negative imaginary part, etc.)? Would be great to identify those and add a comment accordingly in the implementation and the tests

Reply:
It is, but in the time since I made that comment I discovered that jax's complex number maths is buggy in general. It seems to cause problems when the values are too big or too small, but not anywhere near breaking floating point maths or anything, just stuff like stuff like `log1p(-1.-0.00110054j)` or `arccos(-35246690.-2.j)` give the wrong answers. Not sure what the best way to deal with it is since I'm pretty sure it's a bug on their side

Reply:
got it! ig it would be great if you could create an issue on the jax repo pointing out to some of these bugs and then add that link to this comment, just for reference. Thanks!

Reply:
thanks!